src/core/geometry/qgsrectangle.h

```diff
@@ -43,7 +43,12 @@ class CORE_EXPORT QgsRectangle
   public:
 
     //! Constructor for a null rectangle
-    QgsRectangle() = default; // optimised constructor for null rectangle - no need to call normalize here
+    QgsRectangle()
+      : mXmin( std::numeric_limits<double>::max() )
+      , mYmin( std::numeric_limits<double>::max() )
+      , mXmax( -std::numeric_limits<double>::max() )
+      , mYmax( -std::numeric_limits<double>::max() )
```

Comment:
I'm wondering if it's cleaner to just set these all as nan, and adjust the isnull test accordingly. What's your thoughts?